models/roles.js

```diff
@@ -0,0 +1,31 @@
+var roleModel
+
+function start(options) {
+  const { schema } = options
+  /** role storage model */
+  roleModel = schema.define('roles', {
+    name: { type: String, length: 64 }
+  })
+}
+
+module.exports = {
+  start: start,
+  getRoleIdByName: (name) => {
+    return new Promise((resolve, rej) => {
+      roleModel.find({ where: { name } }, function(err, roles) {
+        if (err) {
+          console.error(err)
+          return rej(err)
+        }
+        if (roles.length != 1) {
+          return cb('length ' + roles.length, false)
+        }
+        //cb(err, roles[0].id)
+        resolve(roles[0].id)
```

Comment:
Are we sure that roles will always be of length >== 1 here?
Will get undefined error otherwise

Reply:
yup, the previous if makes sure of that, length will always exactly one here.